src/components/ModalPortal.js

```diff
@@ -137,8 +138,23 @@ export default class ModalPortal extends Component {
   afterClose = () => {
     // Remove body class
     bodyClassList.remove(this.props.bodyOpenClassName);
-    focusManager.returnFocus();
-    focusManager.teardownScopedFocus();
+
+    if (this.shouldReturnFocus()) {
+      focusManager.returnFocus();
+      focusManager.teardownScopedFocus();
+    }
+  };
+
+  shouldReturnFocus = () => {
+    // Don't restore focus to the element that had focus prior to
+    // the modal's display if:
+    // 1. Focus was never shifted to the modal in the first place
+    //    (shouldFocusAfterRender = false)
+    // 2. Explicit direction to not restore focus
+    return (
+      this.props.shouldFocusAfterRender ||
+      this.props.shouldReturnFocusAfterClose
```

Comment:
I believe this should be `&&` not `||`. Currently I have to pass both `shouldFocusAfterRender={false}` and `shouldReturnFocusAfterClose={false}` to opt out of refocusing.

Reply:
Looking at this now, if `shouldReturnFocusAfterClose=false` would also disable the 'focus after render'. They should be able to be independent of each other.

Reply:
NOTES:

If `shouldFocusAfterRender=true and shouldReturnFocusAfterClose=false`, we put the element in the focus manager and on close we just pop it out.

`shouldFocusAfterRender=false and shouldReturnFocusAfterClose=true` must be noop just like `shouldFocusAfterRender=false and shouldReturnFocusAfterClose=false`.

`shouldFocusAfterRender=true and shouldReturnFocusAfterClose=true` is the default behavior.

Does this make sense?

Reply:
Yeah, sounds reasonable

Reply:
Btw, what are the usecases for `shouldFocusAfterRender`? Looks like it just focuses a [div](https://github.com/reactjs/react-modal/blob/master/src/components/ModalPortal.js#L310-L317) without any keyboard handlers

Reply:
To set the scope for tabbing, so it would leave the modal until it is open.